src/formatDate.js

```diff
@@ -8,7 +8,39 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateComponents = date.split(fromFormat[3]);
+  const formattedDateComponents = Array(3);
+
+  const indexYY = fromFormat.indexOf('YY');
+  const indexYYYY = fromFormat.indexOf('YYYY');
+  const toIndexYY = toFormat.indexOf('YY');
+  const toIndexYYYY = toFormat.indexOf('YYYY');
+
+  if (indexYY !== -1 || indexYYYY !== -1) {
+    const isFromYY = indexYY !== -1;
+    const isToYY = toIndexYY !== -1;
+    const fromIndex = isFromYY ? indexYY : indexYYYY;
+    const toIndex = isToYY ? toIndexYY : toIndexYYYY;
+    let year = dateComponents[fromIndex];
```

Comment:
In order to improve readability, I'd advise to split input and output years to separate constants/variables with corresponding names:
```
const fromYear = dateComponents[fromIndex];
let toYear = fromYear;
if (isFromYY && !isToYY) {
  toYear = fromYear < 30 ? '20' + fromYear : '19' + fromYear;
}
...

```